momentGW/thc.py

```diff
@@ -0,0 +1,270 @@
+        Xip = coll[: self.nocc, :]
+        Xap = coll[self.nocc :, :]
+        self._blocks["coll"] = coll
+        self._blocks["cou"] = cou
+        self._blocks["Xip"] = Xip
+        self._blocks["Xap"] = Xap
+
+    @property
+    def Coll(self):
+        """
+        Return the (aux, MO) array.
+        """
+        return self._blocks["coll"]
+
+    @property
+    def Cou(self):
+        """
+        Return the (aux, aux) array.
+        """
+        return self._blocks["cou"]
+
+    @property
+    def Xip(self):
+        """
+        Return the (aux, W occ) array.
+        """
+        return self._blocks["Xip"]
+
+    @property
+    def Xap(self):
+        """
+        Return the (aux, W vir) array.
+        """
+        return self._blocks["Xap"]
+
+    @property
+    def nmo(self):
+        """
+        Return the number of MOs.
+        """
+        return self.mo_coeff.shape[-1]
+
+    @property
+    def nocc(self):
+        """
+        Return the number of occupied MOs.
+        """
+        return np.sum(self.mo_occ > 0)
+
+    @property
+    def naux(self):
+        """
+        Return the number of auxiliary basis functions, after the
+        compression.
+        """
+        return self.Cou.shape[0]
+
+
+class TDA:
+    """
+    Compute the self-energy moments using THC integrals in TDA
+
+    Parameters
+    ----------
+    tda: TDA
+        TDA object
+    """
+
+    def __init__(
+        self,
+        gw,
+        nmom_max,
+        integrals,
+        mo_energy=None,
+        mo_occ=None,
+    ):
+        self.gw = gw
+        self.integrals = integrals
+        self.naux = self.integrals.naux
+        self.nmo = self.integrals.nmo
+        self.nmom_max = nmom_max
+        self.total_nmom = self.nmom_max + 1
+        self.nocc = self.integrals.nocc
+
+        self.XiP = self.integrals.Xip
+        self.XaP = self.integrals.Xap
+        self.Z = self.integrals.Cou
+
+        # Get the MO energies for G and W
+        if mo_energy is None:
+            self.mo_energy_g = self.mo_energy_w = gw._scf.mo_energy
+        elif isinstance(mo_energy, tuple):
+            self.mo_energy_g, self.mo_energy_w = mo_energy
+        else:
+            self.mo_energy_g = self.mo_energy_w = mo_energy
+
+        # Get the MO occupancies for G and W
+        if mo_occ is None:
+            self.mo_occ_g = self.mo_occ_w = gw._scf.mo_occ
+        elif isinstance(mo_occ, tuple):
+            self.mo_occ_g, self.mo_occ_w = mo_occ
+        else:
+            self.mo_occ_g = self.mo_occ_w = mo_occ
+
+        # Options and thresholds
+        self.report_quadrature_error = True
+        if "ia" in getattr(self.gw, "compression", "").split(","):
+            self.compression_tol = gw.compression_tol
+        else:
+            self.compression_tol = None
+
+        self.ea = self.mo_energy_w[self.mo_occ_w == 0]
+        self.ei = self.mo_energy_w[self.mo_occ_w > 0]
+
+    def kernel(self):
```

Comment:
You can remove this entire `kernel` function, once the inheritance is fixed and the functions are renamed as stated below it will work perfectly fine with the non-THC `TDA.kernel`